qiskit_ibm_runtime/debug_tools/neat.py

```diff
@@ -0,0 +1,228 @@
+    classically efficiently, either in ideal conditions or in the presence of noise. The
+    simulations' results can be seamelessly compared with other simulation results or with
+    primitive results to draw custom figures of merit.
+
+    .. code::python
+
+        # Initialize a Neat object
+        analyzer = Neat(backend)
+
+        # Map arbitrary PUBs to Clifford PUBs
+        cliff_pubs = analyzer.to_clifford(pubs)
+
+        # Calculate the expectation values in the absence of noise
+        r_ideal = analyzer.simulate(cliff_pubs, with_noise=False)
+
+        # Calculate the expectation values in the presence of noise
+        r_noisy = analyzer.simulate(cliff_pubs, with_noise=True)
+
+        # Calculate the expectation values for a different noise model
+        analyzer.noise_model = another_noise_model
+        another_r_noisy = analyzer.simulate(cliff_pubs, with_noise=True)
+
+        # Run the Clifford PUBs on a QPU
+        r_qpu = estimator.run(cliff_pubs)
+
+        # Calculate useful figures of merit using mathematical operators, for example the relative
+        # difference between experimental and noisy results, ...
+        rel_diff = abs(r_noisy[0] - r_qpu[0]) / r_noisy[0]
+
+        # ... the signal-to-noise ratio between experimental and ideal results, ...
+        ratio = r_qpu[0] / r_ideal[0]
+
+        # ... or the absolute difference between results obtained with different noise models
+        abs_diff = abs(r_noisy[0] - another_r_noisy[0])
+
+    Args:
+        backend: A backend.
+        noise_model: A noise model for the operations of the given backend. If ``None``, it
+            defaults to the noise model generated by :meth:`NoiseModel.from_backend`.
+    """
+
+    def __init__(self, backend: Backend, noise_model: Optional[NoiseModel] = None) -> None:
+        if not HAS_QISKIT_AER:
+            raise ValueError(
+                "Cannot initialize object of type 'Neat' since 'qiskit-aer' is not installed. "
+                "Install 'qiskit-aer' and try again."
+            )
+
+        self._backend = backend
+        self.noise_model = (
+            noise_model
+            if noise_model is not None
+            else NoiseModel.from_backend(backend, thermal_relaxation=False)
+        )
+
+    @property
+    def noise_model(self) -> NoiseModel:
+        r"""
+        The noise model used by this analyzer tool for the noisy simulations.
+        """
+        return self._noise_model
+
+    @noise_model.setter
+    def noise_model(self, value: NoiseModel) -> NoiseModel:
+        """Sets a new noise model.
+
+        Args:
+            value: A new noise model.
+        """
+        self._noise_model = value
+
+    def backend(self) -> Backend:
+        r"""
+        The backend used by this analyzer tool.
+        """
+        return self._backend
+
+    def simulate(
+        self,
+        pubs: Sequence[EstimatorPubLike],
+        with_noise: bool = True,
```

Comment:
This is a matter of subjective preference, and I'm happy enough with the current setup, but I suspect it would feel slightly more ergonomic to me to split this into two methods along this bool:

```python
great_ratio = neat.ideal_sim(puberoos) / neat.noisy_sim(puberoos)
```

Reply:
Not the first person to suggest this, I am happy to go for it